### DISCUSSION TITLE
add support for `switch-case` output

### DISCUSSION THREAD

**Discussion by ivanhofer** [ivanhofer/typesafe-i18n#135] [Ideas]
Similar tho the [`{select} format`](https://formatjs.io/docs/core-concepts/icu-syntax/#select-format) in the ICU-syntax, `typesafe-i18n` should be able to handle `switch-case` options.
Main use-case is something like gender-related output, but it could be used for uther use-cases too.

I propose this could look like:

#### named keys

`{0}`, `{1}`, `{2}`, ...

```ts
const en = {
   response: '{gender|[male:He,female:She,other:They]} will respond shortly.'
}

LL.response({ gender: 'male' }) => 'He will respond shortly'
```

#### index-based keys

`{key}`, `{key2}`, ...

```ts
const en = {
   response: '{0|[male:He,female:She,other:They]} will respond shortly.'
}

LL.response('female') => 'She will respond shortly'
```

####  better readability trough spacing

Spacing is allowed and will be trimmed before inserting it into the sentence.

```ts
const en = {
   response: '{0 | [ male: He, female: She, other: They ] } will respond shortly.'
}

LL.response('other') => 'They will respond shortly'
```

####  default case

 A `default-case` should be marked with `*`

```ts
const en = {
   response: '{0 | [ male: He, female: She, other: They ] } will respond shortly.',
   response2: '{0 | [ male: He, female: She, *: They ] } will respond shortly.'
}

LL.response('other') => 'They will respond shortly'
LL.response(null) // => TypeScript Error

LL.response2('other') => 'They will respond shortly'
LL.response2(null) // => 'They will respond shortly'
```

#### formatter chaining

It should be possible to chain it and pass it to another formatter

```ts
const en = {
   response: '{gender | [ male: He, female: She, other: They ] | uppercase } will respond shortly.'
}

LL.response({ gender :'male' }) => 'HE will respond shortly'
```

#### typesafety

This should be implemented in a typesafe way, so you would also need to pass the same options for all other translations.

```ts
const en = {
   response: '{gender | [ male: He, female: She, other: They ] | uppercase } will respond shortly.'
}
const de = {
   response: '{gender | [ male: Er wird, female: Sie wird, other: Sie werden ]} in Kürze antworten.'
}
const it = {
   response: '{gender | [ male: Lui, female: Lei ]} risponderà a breve.' // => TypeScript Error: 'other' is missing
}
```

**Comment by abastardi** [comment]
This looks really good. One minor aesthetic suggestion -- as it is common for braces to represent objects/maps and brackets to represent lists/arrays, maybe wrap the key/value map in braces rather than brackets:

```
{ 0 | { male: He, female: She, other: They } }
```

**Reply by ivanhofer** [reply]
I wasn't sure about that first, so I have chosen the array brackets. But the more I think about it I also prefer the object syntax. Thanks for your feedback

**Comment by ivanhofer** [comment]
added in version `2.55.0`
